File: 2022/ocaml/bin/main.ml
(* dune exec ./bin/main.exe *)
let () =
  let open Aoc_lib in
  let input = Utils.read_input "../_inputs/day01" |> Day01.from_input in
  let answer1 = input |> Day01.most_calories in
  let answer2 = input |> Day01.top3_calories in
  Printf.printf "\n\nPart 1 answer is: %d\n" answer1;
  Printf.printf "Part 2 answer is: %d\n" answer2;

  (* day 2*)
  let input = Utils.read_input "../_inputs/day02" |> Day02.from_input in
  let answer1 = Day02.rounds input |> Day01.sum in
  Printf.printf "\n\nDay2 part 1 answer is: %d\n" answer1;
  let answer2 = Day02.rounds_v2 input |> Day01.sum in
  Printf.printf "nDay2 part 2 answer is: %d\n" answer2

File: 2022/haskell/app/Main.hs
module Main where

import qualified Day01
import System.IO (IOMode (ReadMode), hClose, hGetContents, openFile)

main :: IO ()
main = do
    handle <- openFile "../inputs/day01" ReadMode
    input <- hGetContents handle
    let input' = Day01.fromInput input
    let answer1 = Day01.greatestSum input'
    putStrLn $ "Answer 1 is: " ++ show answer1
    let answer2 = Day01.top3Calories input'
    putStrLn $ "Answer 2 is: " ++ show answer2
    hClose handle
